crates/router/src/core/admin.rs

```diff
@@ -885,21 +883,16 @@ impl MerchantAccountUpdateBridge for api::MerchantAccountUpdate {
 
         // This supports changing the business profile by passing in the profile_id
         let business_profile_id_update = if let Some(ref profile_id) = self.default_profile {
-            if !profile_id.is_empty_after_trim() {
-                // Validate whether profile_id passed in request is valid and is linked to the merchant
-                core_utils::validate_and_get_business_profile(
-                    state.store.as_ref(),
-                    key_manager_state,
-                    key_store,
-                    Some(profile_id),
-                    merchant_id,
-                )
-                .await?
-                .map(|business_profile| Some(business_profile.profile_id))
-            } else {
-                // If empty, Update profile_id to None in the database
-                Some(None)
-            }
```

Comment:
We won't be able to allow the "if empty set default profile to null" behavior anymore if using the `ProfileId` domain type for the field, since the domain type doesn't allow constructing profile IDs from empty strings.

Reply:
Thanks for pointing this out!

Reply:
@JeevaRamu0104 Tagging for visibility